server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -487,6 +487,105 @@ public void testExplain() throws Exception {
+                    .field("type", "nested")
+                    .endObject()
+                    .startObject("nested2")
+                    .field("type", "nested")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+        );
+
+        ensureGreen();
+
+        for (int i = 0; i < 3; i++) {
+            client().prepareIndex("test")
+                .setId(String.valueOf(i))
+                .setSource(
+                    jsonBuilder().startObject()
+                        .field("field" + i, "value" + i)
+                        .startArray("nested" + i)
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .endArray()
+                        .endObject()
+                )
+                .setRefreshPolicy(IMMEDIATE)
+                .get();
+        }
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs"));
+    }
+
+    public void testExplainSingleDocIndexingWithDeletedDoc() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping(
+                jsonBuilder().startObject()
+                    .startObject("properties")
+                    .startObject("nested1")
+                    .field("type", "nested")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+        );
+
+        ensureGreen();
+
+        client().prepareIndex("test")
+            .setId("1")
+            .setSource(
+                jsonBuilder().startObject()
+                    .field("field1", "value1")
+                    .startArray("nested1")
+                    .startObject()
+                    .field("n_field1", "n_value1")
+                    .endObject()
+                    .startObject()
+                    .field("n_field1", "n_value1")
+                    .endObject()
+                    .endArray()
+                    .endObject()
+            )
+            .setRefreshPolicy(IMMEDIATE)
+            .get();
+
+        // this will lead to deleted docs
         indexRandomForConcurrentSearch("test");
```

Comment:
I think we should use `indexRandomForMultipleSlices` instead here to add deleted docs even in non-concurrent search path. That way test will be validating explain output for deleted docs for both concurrent/non-concurrent flow.

Reply:
As explained below, I don't think we need 3 different test cases, we can keep one test case as before and add `indexRandomForMultipleSlices` and remove the validation for the range as that can vary.